sophus/rxso2.hpp

```diff
@@ -503,26 +495,20 @@ class RxSO2 : public RxSO2Base<RxSO2<Scalar_, Options>> {
   ///
   SOPHUS_FUNC static Sophus::Matrix<Scalar, num_parameters, DoF> Dx_exp_x(
       Tangent const& a) {
-    using std::cos;
     using std::exp;
-    using std::sin;
     Scalar const theta = a[0];
     Scalar const sigma = a[1];
 
-    Sophus::Matrix<Scalar, num_parameters, DoF> J;
-    J << -sin(theta), cos(theta), cos(theta), sin(theta);
-    return J * exp(sigma);
+    return SO2<Scalar>::exp(theta).matrix().rowwise().reverse() * exp(sigma);
```

Comment:
One line comment why "rowwise.reverse" does the right thing here.

(Maybe writing down that what we want is:

```
-sin(t) cos(t)
cos(t) sin(t)
```